wrappers/react/src/hotTable.tsx

```diff
@@ -215,17 +215,30 @@ class HotTable extends React.Component<HotTableProps, {}> {
    * @param {React.ReactElement} rendererElement React renderer component.
    * @returns {Handsontable.renderers.Base} The Handsontable rendering function.
    */
-  getRendererWrapper(rendererElement: React.ReactElement): typeof Handsontable.renderers.BaseRenderer | any {
+  getRendererWrapper(rendererElement: React.ReactElement): typeof Handsontable.renderers.BaseRenderer {
     const hotTableComponent = this;
 
+
+    
+    // Map of the TD elements to their portal containers, for reuse instead of creating a new portal container every 
+    // render, for every cell. This reduces flashes of empty cells when re-rendering.
+    // TODO upgrade typescript to be able to use WeakRef in case the HTMLElement could be GC'd before the renderer wrapper
+    const portalContainerCache = new Map<string, HTMLElement>();
+
     return function (instance, TD, row, col, prop, value, cellProperties) {
       const renderedCellCache = hotTableComponent.getRenderedCellCache();
 
-      if (renderedCellCache.has(`${row}-${col}`)) {
-        TD.innerHTML = renderedCellCache.get(`${row}-${col}`).innerHTML;
+      const key = `${row}-${col}`;
+
+      if (renderedCellCache.has(key)) {
+        TD.innerHTML = renderedCellCache.get(key).innerHTML;
       }
 
       if (TD && !TD.getAttribute('ghost-table')) {
+        // prevent table clones sharing portal containers by including the table clone's className in the key
+        const portalKey = `${TD.closest(".handsontable")?.className}__${key}`
```

Comment:
Have you tried `.ht_clone_master` class?

Reply:
@trey-miller any blocker for this PR?

Reply:
Hi @rmlevangelio , can you explain how you are thinking of using that class? Right now it finds the closest ancestor with `.handsontable` class, and uses the rest of the className on that element for the unique `portalKey`. The `.ht_clone_master` class is one of the differentiating classes, so if the TD is within that it will have one portal key, and the same row+col within another clone will have another portal key, so they don't use the same portal container.

I think it could be improved if there was a different way to differentiate which "clone" the TD is in to store unique portal containers, but I'm not sure if there is any other way available currently. It seems to work fine this way, I just was thinking it could be better if it didn't need to traverse the DOM parents to find out every time.

The main blocker I have though is that I'm still waiting for approval to sign the contributor agreement. I'm hoping that will be done soon.

Reply:
I tried it on my end and yes, I think your solution is the only way for now to get the container.

Reply:
Hey, you should be able to do this instead  `${instance.guid}__${key}` instead of `${TD.closest(".handsontable")?.className}__${key}` the guid is unique per instance so should be what you are looking for

Reply:
Hey @trey-miller did you manage to get your contributor agreement signed?